storage/innobase/row/row0log.cc

```diff
@@ -3202,10 +3417,23 @@ row_log_allocate(
 	log->path = path;
 	log->n_old_col = index->table->n_cols;
 	log->n_old_vcol = index->table->n_v_cols;
+	log->crypt_tail = log->crypt_head = NULL;
+	log->index = index;
 
 	dict_index_set_online_status(index, ONLINE_INDEX_CREATION);
 	index->online_log = log;
 
+	if (log_tmp_is_encrypted()) {
+		ulint size = srv_sort_buf_size;
+		log->crypt_head = static_cast<byte *>(os_mem_alloc_large(&size));
+		log->crypt_tail = static_cast<byte *>(os_mem_alloc_large(&size));
+
+		if (!log->crypt_head || !log->crypt_tail) {
```

Comment:
Doesn't log->crypt_head != NULL, log->crypt_tail == NULL result in a memory leak for the former?

Reply:
`row_log_free` will deallocate both as I read